src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -121,9 +122,16 @@ class ReferralController(
   @GetMapping("/sent-referrals/summary/service-provider")
   fun getServiceProviderSentReferralsSummary(
     authentication: JwtAuthenticationToken,
+    @Nullable @RequestParam(name = "dashboardType", required = false) dashboardTypeSelection: String?,
   ): List<ServiceProviderSentReferralSummaryDTO> {
     val user = userMapper.fromToken(authentication)
-    return referralService.getServiceProviderSummaries(user).map { ServiceProviderSentReferralSummaryDTO.from(it) }
+    var dashboardType: DashboardType?
+    try {
+      dashboardType = dashboardTypeSelection?.let { DashboardType.valueOf(it) }
+    } catch (e: IllegalArgumentException) {
+      throw ResponseStatusException(HttpStatus.BAD_REQUEST, "unknown dashboardType [dashboardType=$dashboardTypeSelection]")
```

Comment:
Should we use a customer converter for this?

Reply:
Do you mean having an exception handler defined in errorConfiguration for IllegalArgumentException?

Reply:
```
  @ExceptionHandler(IllegalArgumentException::class)
  fun handleIllegalArgumentException(e: IllegalArgumentException): ResponseEntity<ErrorResponse> {
    logger.info("Illegal argument provided", e)
    return errorResponse(HttpStatus.BAD_REQUEST, "Illegal argument provided", e.message, userMessageForWebClientException(HttpStatus.BAD_REQUEST))
  }
```

Reply:
I've added this exception handler